### ISSUE TITLE
split-init interaction with warning about local-from-distributed

### ISSUE THREAD

**Issue by mppf** [chapel-lang/chapel#14746]
How should split init interact with the warning about initializing a non-distributed domain from a distributed one?
Previously this warning relied on the difference between initialization and assignment. The warning was added in PR #10791.

For example:

``` chapel
// copy to a default-rectangular array, for writeln()
proc copyToDF(A:[]) {
  var D: domain(A.rank, A.domain.idxType, A.domain.stridable);
  D = A.domain; // warning here if this is now initialization
  var Res: [D] A.eltType = A;
  return Res;
}
```

```
warning: initializing a non-distributed domain from a distributed domain. If you didn't mean to do that, add a dmapped clause to the type expression or remove the type expression altogether
```

When this is resolved, update these tests, as PR #14691 adds a workaround to them:
 * test/distributions/dm/s7.chpl 
 * test/distributions/dm/s8.chpl 
 * test/distributions/dm/s9.chpl 
 * test/distributions/ferguson/init-local-from-distributed-warning.chpl 
 * test/studies/hpcc/HPL/bradc/hpl-blc-noreindex.chpl 
 * distributions/robust/arithmetic/basics/test_domain_align.chpl

**Comment by mppf** [comment]
Brainstorming some ideas to resolve this:
1.  make the error only apply to non-split init (but that would require compiler changes)
2. accept the warning for the pattern demonstrated above
    a. tests use the workaround of mentioning the variable before initializing it
    b. tests include the warning in .good files
    c. tests are updated to call a new library routine to copy to default-rectangular
3. extend the warning to apply also to assignment

**Comment by bradcray** [comment]
I'd probably do 2b myself.  Could you save me some time to reproduce myself by pointing to where the warning would come from in `test/studies/hpcc/HPL/bradc/hpl-blc-noreindex.chpl` and `distributions/robust/arithmetic/basics/test_domain_align.chpl` (the two files I'm most familiar with in your list).

**Comment by mppf** [comment]
Sure.
https://github.com/chapel-lang/chapel/blob/921f541b2ae19cc55d951b93d5295c73e4f9887c/test/studies/hpcc/HPL/bradc/hpl-blc-noreindex.chpl#L186-L189

https://github.com/chapel-lang/chapel/blob/921f541b2ae19cc55d951b93d5295c73e4f9887c/test/distributions/robust/arithmetic/basics/test_domain_align.chpl#L42-L44

**Comment by bradcray** [comment]
Thanks for pointing to the code segments.  I don't think either of those change my answer (and I provided a bit more rationale for my answer in an edit just above, capturing some of our offline discussion).

**Comment by mppf** [comment]
I'll plan to do 2b for the moment.

**Comment by mppf** [comment]
Addressed (for now at least) by PR  #14802 which does 2b.
